app/assets/javascripts/application.js

```diff
@@ -0,0 +1,45 @@
+'use strict'
+
+document.onreadystatechange = function () {
+  if (document.readyState === 'interactive') {
+    var editDocumentForm = document.querySelector('.edit_document')
+    if (editDocumentForm == null) { return } // only run on edit document page
+    var urlPreview = document.getElementById('url-preview-id')
+    var basePath = document.getElementById('base-path-id')
+    var documentTitle = document.getElementById('document-title-id')
+    var noTitle = document.getElementById('no-title-id')
+    var errorGeneratingPath = document.getElementById('error-generating-path-id')
+    var showErrorMessage = function () {
+      urlPreview.setAttribute('class', 'app-hidden')
+      noTitle.setAttribute('class', 'app-hidden')
+      errorGeneratingPath.removeAttribute('class')
+    }
+    documentTitle.onblur = function () {
+      var url = new URL(document.location.origin + editDocumentForm.getAttribute('data-generate-path-path'))
+      if (!documentTitle.value) {
+        noTitle.removeAttribute('class')
+        urlPreview.setAttribute('class', 'app-hidden')
+        errorGeneratingPath.setAttribute('class', 'app-hidden')
+        return
+      }
+      url.searchParams.append('title', documentTitle.value)
+      window.fetch(url, {
+        credentials: 'include',
+        headers: {
+          'Content-Type': 'application/json; charset=utf-8'
+        }
+      }).then(function (response) {
+        if (!response.ok) {
+          throw Error('Unable to generate response.')
+        }
+        response.json().then(function (result) {
+          noTitle.setAttribute('class', 'app-hidden')
```

Comment:
Since we have a showErrorMessage function, can we have a hideErrorMessage function as well? I think that's what this is doing.

Reply:
this is showing the URL preview, it only happens the one time so I don't feel like extracting it benefits us by much? Happy to compromise if you disagree.